### PR TITLE
Enable TCK to run on COSC (and other potential implementations)

### PR DESCRIPTION
**Pull request by s1ck**
This PR generalizes our TCK testing infrastructure to be able to run on different engines with different blacklists etc.. At the moment we run on Spark and COSC.

### CODE REVIEW

File: caps-tck/src/test/scala/org/opencypher/caps/tck/TCKCOSCTest.scala [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import org.opencypher.caps.cosc.test.COSCTestSuite
+import org.opencypher.caps.cosc.test.support.creation.cosc.COSCTestGraphFactory
+import org.opencypher.caps.test.support.creation.TestGraphFactory
+import org.opencypher.tools.tck.api.CypherTCK
+import org.scalatest.Tag
+import org.scalatest.prop.TableDrivenPropertyChecks._
+
+import scala.util.{Failure, Success, Try}
+
+class TCKCOSCTest extends COSCTestSuite {
```
Comment (Mats-SX) [inline caps-tck/src/test/scala/org/opencypher/caps/tck/TCKCOSCTest.scala]: I lol'd

Comment (Mats-SX) [inline caps-tck/src/test/scala/org/opencypher/caps/tck/TCKCOSCTest.scala]: I lol'd

Comment (s1ck) [reply caps-tck/src/test/scala/org/opencypher/caps/tck/TCKCOSCTest.scala]: me 2

**Comment by pstutz** [comment]
The TCK test output from COSC seems to make our log too long to display conveniently on Travis CI; we should probably reduce it.

**Comment by s1ck** [comment]
Addressed in https://github.com/opencypher/cypher-for-apache-spark/pull/289

**Pull request merged**